### PR TITLE
Dockerize bedrock-setup

### PR DESCRIPTION
**Pull request by msporny**
This PR adds a Dockerfile so that we can release this package as a docker image on dockerhub. This branch has been built, released, and tested as a part of the Veres One PoC Node AWS build/setup process:

https://hub.docker.com/layers/digitalbazaar/bedrock-setup/2.0.0/images/sha256-8b5a03f13c37ba8b6c92cf4e56aaff8a6664d936d8bd2183536266f734a76d2a

It does not currently optimize the built image. Ideally, we'd do that, but don't know how that is impacted by @davidlehn's latest webpack updates.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
   "less": "app.less",
   "scripts": {
     "start": "node --preserve-symlinks setup.localhost.js",
-    "postinstall": "node --preserve-symlinks setup.localhost.js compile-less"
+    "compile-less": "node --preserve-symlinks setup.localhost.js compile-less"
```
Comment (msporny) [inline package.json]: @mattcollier, Is this wrong?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+RUN mkdir -p /etc/bedrock-setup
+RUN chown -R node:node /home/node/app
+WORKDIR /home/node/app
+COPY --chown=node:node . .
+
+FROM base AS build
+RUN apk add --no-cache git bash
+WORKDIR /home/node/app
+USER node
+RUN npm install --no-optional --production
```
Comment (mattcollier) [inline Dockerfile]:
```suggestion
RUN npm ci --no-optional --production
```

File: package.json (hunk 1 above)
Comment (mattcollier) [reply package.json]: Wrong in what way?  Is `npm run compile-less` being executed somewhere now?

**Review by mattcollier — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (davidlehn) [reply package.json]: `compile-less` doesn't exist with the newer views/webpack code.

**Comment by davidlehn** [comment]
When the new packages I worked on get merged in, there will be some small changes.
- "compile-less" doesn't exist anymore.
- There's a "bundle" command (vs optimize).
- There's a new config value (or command line flag if that's easier) to use the production output.
- Also production systems should be run with `NODE_ENV="production"`.

File: package.json (hunk 1 above)
Comment (msporny) [reply package.json]: @davidlehn I don't understand what I'm supposed to do here given the information you've put above. Is this something you're going to take care of, or should I take care of it?

**Review by msporny — COMMENTED** [review]
Will move this over to the new ci-based approach and make a new docker release.

File: package.json (hunk 1 above)
Comment (msporny) [reply package.json]: I guess I can try to update this to bedrock@3?

**Review by mattcollier — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (msporny) [reply package.json]: Got rid of all compile-less/postinstall garbage and upgraded to bedrock@3 in 3f8bc3488c72f9cf0ec23222174b3a32565f4cf5.

**Comment by msporny** [comment]
@davidlehn wrote:
> "compile-less" doesn't exist anymore.

Removed from package.json file.

> There's a "bundle" command (vs optimize).

Using this now in the Dockerfile command.

> There's a new config value (or command line flag if that's easier) to use the production output.

Did this, among other undocumented things to get everything working in production mode.

> Also production systems should be run with NODE_ENV="production".

Done in 3f8bc3488c72f9cf0ec23222174b3a32565f4cf5.

**Comment by msporny** [comment]
Merging down to master... don't know if this is a breaking release, or even how to release this package. Can someone help do that, please?

**Pull request merged**